lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -458,11 +461,26 @@ private boolean achievedWithGpg45Profile(
                         ? gpg45ProfileEvaluator.getFirstMatchingProfile(
                                 gpg45Scores, requestedVot.getSupportedGpg45Profiles())
                         : Optional.empty();
+        List<SignedJWT> gpg45Credentials = new ArrayList<>();
+        for (SignedJWT credential : credentials) {
+            if (!VcHelper.isOperationalProfileVc(credential)) {
+                gpg45Credentials.add(credential);
+            }
+        }
 
         // Successful match
         if (matchedGpg45Profile.isPresent()) {
+            // remove weaker operational profile
+            if (configService.enabled(INHERITED_IDENTITY.getName())) {
```

Comment:
Might be worth considering adding a check that the requested Vot is P2. If we add P1 this'll be important.

Reply:
We only want to remove the operational profiles if the user has attained a stronger profile, which P2 is. P1 however isn't, so in a future scenario where we handle P1s, we wouldn't want to remove the operational profiles.